### PR TITLE
Configure log level and log-color via commandline

### PR DESCRIPTION
**Pull request by cfconrad**
Introduce new parameter options `--loglevel` and `--log-color`

### CODE REVIEW

File: qatrfm/utils/logger.py [hunk 1]
```diff
@@ -11,57 +11,66 @@
...
+        return super().__init__(logger_name)
+
+    @staticmethod
+    def colorize(msg, color):
+        COLORS_MAP = {
+                'blue': 34, 'green': 32, 'red': 31, 'yellow': 33,
+                'lightgrey': 37
+                }
+        if color in COLORS_MAP.keys():
+            return "\033[{}m{}\033[0m".format(COLORS_MAP[color], msg)
```
Comment (jlausuch) [inline qatrfm/utils/logger.py]:
```suggestion
            return "\033[1;{}m{}\033[0m".format(COLORS_MAP[color], msg)
```

I'd like to keep the colored messages bold. They don't look so nice (at least on my prompt) with light colors.
http://imagebin.suse.de/2706/img

@asmorodskyi  WDYT?

**Comment by jlausuch** [comment]
Much better than exporting LOG_COLORS in environment. I just have a small question about aesthetics.. bold vs light.

Interestingly, Terraform CLI does actually the opposite, by default colours are enabled. Anyway, I prefer having them disabled by default because this tool will be mainly used in automated environments.

File: qatrfm/utils/logger.py (hunk 1 above)
Comment (asmorodskyi) [reply qatrfm/utils/logger.py]: I would use something like http://lnav.org/. So I don't see use case for this option

File: qatrfm/cli.py [hunk 2]
```diff
@@ -76,9 +76,15 @@ def find_tf_file(basedir):
 @click.option('--no-clean', 'no_clean', is_flag=True,
               help="Don't clean the environment when the tests finish. "
               "This is useful for debug and troubleshooting.")
-def cli(test, path, tfvar, snapshots, no_clean):
+@click.option('--loglevel', 'loglevel', type=click.Choice([
+              'CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG']),
```
Comment (asmorodskyi) [inline qatrfm/cli.py]: why reinvent the wheel ? https://docs.python.org/3.1/library/logging.html#logging-levels

Comment (jlausuch) [reply qatrfm/cli.py]: it's not reinventing the wheel. These codes are passed to the python logger later, which is exactly the link you pasted. This just allows the user to select which log level to display.

File: qatrfm/utils/logger.py (hunk 1 above)
Comment (cfconrad) [reply qatrfm/utils/logger.py]: never heard something about inav, sounds interesting. Are you using it somewhere, maybe you can show me.

Regarding bold, I don't have hard feelings, for me bold is good as well.

File: qatrfm/cli.py (hunk 2 above)
Comment (asmorodskyi) [reply qatrfm/cli.py]: I thought something like this https://github.com/pallets/click/issues/605 , but looks like it is not supported yet ...

File: qatrfm/cli.py [hunk 3]
```diff
@@ -76,9 +76,15 @@ def find_tf_file(basedir):
 @click.option('--no-clean', 'no_clean', is_flag=True,
               help="Don't clean the environment when the tests finish. "
               "This is useful for debug and troubleshooting.")
-def cli(test, path, tfvar, snapshots, no_clean):
+@click.option('--loglevel', 'loglevel', type=click.Choice([
```
Comment (asmorodskyi) [inline qatrfm/cli.py]: is it not possible ?
```suggestion
@click.option('--loglevel', 'loglevel', type=click.Choice([logging.loglevel]),
```

File: qatrfm/utils/logger.py (hunk 1 above)
Comment (asmorodskyi) [reply qatrfm/utils/logger.py]: nope I have never use it but I just wanted to show that idea of inserting such functionality inside qatrfm looks strange for me. In my understanding this problem could and should be solved outside. And actually comment from @jlausuch  with screenshots from **his** terminal is clearly showing why :) 
But I have no hard feelings about it after all can just avoid using this option :) Just discussion if it should be bold or not is looks strange :)

Comment (cfconrad) [reply qatrfm/utils/logger.py]: I agree, that using colors has this difficulties.
So for me conclusion is, we will keep it for now. But looking for better way's when needed.

File: qatrfm/cli.py (hunk 3 above)
Comment (cfconrad) [reply qatrfm/cli.py]: yes, if logging would have a attribute `loglevel` which is the list of all possible loglevels.

File: qatrfm/utils/logger.py (hunk 1 above)
Comment (asmorodskyi) [reply qatrfm/utils/logger.py]: ok

File: qatrfm/cli.py (hunk 3 above)
Comment (asmorodskyi) [reply qatrfm/cli.py]: ok

**Comment by asmorodskyi** [comment]
LGTM

**Pull request merged**